mail_notification_with_history/models/mail_message.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class Message(models.Model):
+    _inherit = "mail.message"
+
+    def get_notification_message_history(self):
+        """Get the list of messages to include into an email notification history."""
+        if not self.env[self.model].with_mail_notification_history():
+            return []
+        if not self._with_notification_message_history():
+            return []
+        domain = self._get_notification_message_history_domain()
+        messages = self.env["mail.message"].search(domain, order="date desc")
```

Comment:
How about also implementing a limit? It could be important in case there are long discussions
Could be overrideable with a method `_get_notification_message_history_limit` in the `mail.thread` (see comment below about `_get_notification_message_history_domain`)

Reply:
About the limit of messages lets see with the use of the module if this becomes a requirement.